### ISSUE TITLE
How about BFL devices?

### ISSUE THREAD

**Issue by irungentootoo** [samr7/vanitygen#44]
How about butterflylabs devices, it posible to use in vanity?

**Comment by jgbreezer** [comment]
I've also been curious and just been looking into whether FPGA or ASICs
could be at all used for vanity-gen - they dont use the same logic so ASICs
would be hardest to make work with it, if at all (in case you're not that
technical, ASICs are app-specific as the acronym indicates, and impossible
to make significant changes to their logic).

I was wondering if any of the ASIC designs (ideally the cheaper ones but
unlikely) would be modular enough to work around/only call specific parts
of it, but I strongly suspect not (writing as an experienced C/C++/Python
software dev not a hardware techy though I've done some basics at
university many years ago): probably the ASICs are just too different from
whats needed, even if you could make it believe the "block" to be hashed
was your nonce-incremented address based on the partial public key; setting
the difficulty could be set to the highest possible to match more exactly
but I think its matching the wrong thing and couldn't be made to match the
right one. Disclaimer: I may slightly misunderstand the algorithm, having
only read it a couple of times :)

FPGA's have more potential, and some of them might even support OpenCL so
need less porting effort - or MPI or other relatively high level
languages/systems to program them in. I only own a lower level GPU card -
still 20x better Hashes/sec than my AMD dual-core CPU, but I want faster
without spending well into the hundreds or more and not sure I've got
enough spare 4x/16x/.. PCIe slots :)

I did have more of a read about the (Kickstarter'd) Adapteva Epiphany -
basically a couple(iirc) of ARM cores, Xilinx chips and some FPGAs.
Comparing their CoreMark score to my CPU/badly-scaled-up hypothetical GPU
score (tried to get one, needs work) didn't seem enough value for money,
plus they've been slow/uncommunicative at times and its too late to
pre-order on their site and hope to get one soon.

Shouldn't we move this to bitcointalk.org maybe (there's an existing vg.
thread there too).

**Comment by BlinkyStitt** [comment]
Mining ASICs are useless for vanitygen.  Mining uses sha256 and address generation uses sha256 and ripemd160.

You could program an FPGA, but I don't think anyone has.

The forums would be a better place for this.

**Comment by jgbreezer** [comment]
I'm already aware ASICs are unlikely to allow you to farm out just the
sha-256 portion and report that back to try and speed it up; they'd have to
be more modular to be usable for vanity-generation at all, which would mean
less efficient at blockchain mining (much more data throughput). Which
manufacturer's going to do that when that market is so big and so
competitive that every MHash (or GH now) matters per watt? About none of
them. Vanity-address market is pretty small in comparison so far.

An fpga miner for it exists but its only a proof of concept though it
works. Allegedly; seen one other (than the author) report it working. I'm
hoping to be able to improve it to get notably better than GPU performance
out of it, as currently it seems almost less efficient and a lot slower
than any typical OpenCL-supporting GPU of last couple of years.

I posted to bitcointalk in the existing vanitygen thread, I agree lets keep
replies there from now unless its to do with this code.
